### PR TITLE
Grafana10 support test

### PR DESCRIPTION
**Pull request by tejasshahoracle**

### CODE REVIEW

File: .yarn/cache/d3-color-npm-3.1.0-fc73fe3b15-4931fbfda5.zip
Comment (mamorett) [inline .yarn/cache/d3-color-npm-3.1.0-fc73fe3b15-4931fbfda5.zip:1]: I think .yarn should be in the .gitignore

File: docs/compatmatrix.md
Comment (mamorett) [inline docs/compatmatrix.md:1]: I think there are other docs to update and screenshot to update as well

File: pkg/plugin/query.go
Comment (mamorett) [inline pkg/plugin/query.go:1]: There are still here and there some debug prints

File: src/datasource.ts
Comment (mamorett) [inline src/datasource.ts:1]: There are several functions, which come from metrics plugin, which are not required. For example in metricFindQuery there are many references to template vars which are not used here. And there are functions like getDimensio, getNamespaces which are not used here

File: src/resource.response.parser.ts
Comment (mamorett) [inline src/resource.response.parser.ts:1]: Here also there are metrics related stuff

File: src/types.ts
Comment (mamorett) [inline src/types.ts:1]: here there are many const which make sense only with metric plugins

**Comment by mamorett** [comment]
The OCA Required issue must be fixed, otherwise we cannot merge.

**Review by jvinaya — APPROVED** [review]
LGTM

File: pkg/plugin/query.go
Comment (tejasshahoracle) [reply pkg/plugin/query.go:1]: Resolved

File: .yarn/cache/d3-color-npm-3.1.0-fc73fe3b15-4931fbfda5.zip
Comment (tejasshahoracle) [reply .yarn/cache/d3-color-npm-3.1.0-fc73fe3b15-4931fbfda5.zip:1]: Resolved

File: src/datasource.ts [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+    }
+    const reqBody: JSON = {
+      tenancy: tenancy,
+    } as unknown as JSON;
+    return this.postResource(OCIResourceCall.Compartments, reqBody).then((response) => {
+      return new ResponseParser().parseCompartments(response);
+    });
+  }
+
+  async getNamespacesWithMetricNames(
```
Comment (mamorett) [inline src/datasource.ts:224]: I can still see these functions which are not required in logs plugin

File: src/datasource.ts [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+      compartment: compartment,
+      region: region,
+    } as unknown as JSON;
+    return this.postResource(OCIResourceCall.Namespaces, reqBody).then((response) => {
+      return new ResponseParser().parseNamespacesWithMetricNames(response);
+    });
+  }
+
+
+  async getResourceGroupsWithMetricNames(
```
Comment (mamorett) [inline src/datasource.ts:272]: not required

File: src/plugin.json [hunk 3]
```diff
@@ -18,23 +17,26 @@
...
-    "version": "3.0.0",
-    "updated": "2023-03-01",
+    "version": "5.0.0",
+    "updated": "2023-09-01",
     "screenshots":[
       {
-        "name":"OCI Logs Dashboard",
-        "path":"img/OCI_Logs.png"
+        "name":"OCI Metrics Dashboard",
+        "path":"img/OCI_Metrics.png"
```
Comment (mamorett) [inline src/plugin.json:31]: this is not correct. Be aware this img is mandatory

File: src/resource.response.parser.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    const tenancymodes: string[] = [];
+    if (!results) {
+      return tenancymodes;
+    }
+
+    let rList: string[] = JSON.parse(JSON.stringify(results));
+    return rList;
+  }
+
+  parseCompartments(results: any): OCIResourceItem[] {
```
Comment (mamorett) [inline src/resource.response.parser.ts:59]: to be removed

File: src/resource.response.parser.ts [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    const namespaceWithMetricNames: OCINamespaceWithMetricNamesItem[] = [];
+    if (!results) {
+      return namespaceWithMetricNames;
+    }
+
+    let nmList: OCINamespaceWithMetricNamesItem[] = JSON.parse(JSON.stringify(results));
+    return nmList;
+  }
+
+  parseResourceGroupWithMetricNames(results: any): OCIResourceGroupWithMetricNamesItem[] {
```
Comment (mamorett) [inline src/resource.response.parser.ts:79]: to be removed

File: src/resource.response.parser.ts [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    const rgWithMetricNames: OCIResourceGroupWithMetricNamesItem[] = [];
+    if (!results) {
+      return rgWithMetricNames;
+    }
+
+    let rgList: OCIResourceGroupWithMetricNamesItem[] = JSON.parse(JSON.stringify(results));
+    return rgList;
+  }
+
+  parseDimensions(results: any): OCIResourceMetadataItem[] {
```
Comment (mamorett) [inline src/resource.response.parser.ts:89]: to be removed

File: src/types.ts [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+
+export enum DefaultOCIOptions {
+  ConfigProfile = 'DEFAULT',
+}
+
+export const DEFAULT_TENANCY = "DEFAULT/";
+export const compartmentsQueryRegex = /^compartments\(\s*(\".+\"|\'.+\'|\$\w+)\s*\)|^compartments\(\)\s*/;
+export const regionsQueryRegex = /^regions\(\s*(\".+\"|\'.+\'|\$\w+)\s*\)|^regions\(\)\s*/;
+export const tenanciesQueryRegex = /^tenancies\(\)\s*/;
+export const namespacesQueryRegex = /^namespaces\(\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*(?:,\s*(\".+\"|\'.+\'|\$\w+)\s*)?\)/;
```
Comment (mamorett) [inline src/types.ts:18]: to be removed

File: src/types.ts [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+export const resourcegroupsQueryRegex = /^resourcegroups\(\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*(?:,\s*(\".+\"|\'.+\'|\$\w+)\s*)?\)/;
```
Comment (mamorett) [inline src/types.ts:19]: to be removed

File: src/types.ts [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+export const metricsQueryRegex = /^metrics\(\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*(?:,\s*(\".+\"|\'.+\'|\$\w+)\s*)?\)/;
```
Comment (mamorett) [inline src/types.ts:20]: to be removed

File: src/types.ts [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+export const dimensionQueryRegex = /^dimensions\(\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*,\s*(\".+\"|\'.+\'|\$\w+)\s*(?:,\s*(\".+\"|\'.+\'|\$\w+)\s*)?\)/;
```
Comment (mamorett) [inline src/types.ts:21]: to be removed

File: src/types.ts [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+export const windowsAndResolutionRegex = /^[0-9]+[mhs]$/;
+
+export enum OCIResourceCall {
```
Comment (mamorett) [inline src/types.ts:24]: to be reviewed. In general many of these data structures in this file still have metrics related stuff

Comment (tejasshahoracle) [reply src/types.ts:19]: Done

File: src/resource.response.parser.ts (hunk 4 above)
Comment (tejasshahoracle) [reply src/resource.response.parser.ts]: This is required in getCompartments() in datasource.ts

File: src/datasource.ts (hunk 1 above)
Comment (tejasshahoracle) [reply src/datasource.ts]: Done!

**Review by mamorett — APPROVED** [review]

File: src/datasource.ts [hunk 12]
```diff
@@ -0,0 +1,205 @@
...
+    }
+    const reqBody: JSON = {
+      tenancy: tenancy,
+    } as unknown as JSON;
+    return this.postResource(OCIResourceCall.Regions, reqBody).then((response) => {
+      return new ResponseParser().parseRegions(response);
+    });
+  }
+
+  async getCompartments(tenancy: string): Promise<OCIResourceItem[]> {
```
Comment (mamorett) [inline src/datasource.ts:187]: this guy hjere maybe can be removed

File: src/datasource.ts [hunk 13]
```diff
@@ -0,0 +1,205 @@
...
+  // // **************************** Template variable helpers ****************************
+
+  // /**
+  //  * Matches the regex from creating template variables and returns options for the corresponding variable.
+  //  * Example:
+  //  * template variable with the query "regions()" will be matched with the regionsQueryRegex and list of available regions will be returned.
+  //  */
+  // metricFindQuery?(query: any, options?: any): Promise<MetricFindValue[]> {
+
+  async metricFindQuery?(query: any, options?: any): Promise<MetricFindValue[]> {
```
Comment (mamorett) [inline src/datasource.ts:85]: maybe we can rename to logFindQuery this function

Comment (tejasshahoracle) [reply src/datasource.ts:85]: I tried that and I think it's a  Grafana function and cannot change it

**Review by jvinaya — DISMISSED** [review]

**Review by kz4d1x — APPROVED** [review]
LGTM

**Review by mamorett — APPROVED** [review]

**Review by mamorett — APPROVED** [review]

**Pull request merged**